cli/tests/pcluster/models/test_cluster.py

```diff
@@ -657,6 +663,113 @@ def test_render_and_upload_scheduler_plugin_template(self, mocker, cluster, temp
         )
 
+    @pytest.mark.parametrize(
+        "support_update, instance_type, match",
+        [
+            (
+                "false",
+                "c5.xlarge",
+                "Update failure: The scheduler plugin used for this cluster does not support updating the scheduling "
+                "configuration.",
+            ),
+            ("false", "c5.2xlarge", None),
+            ("true", "c5.xlarge", None),
+        ],
+    )
+    def test_validate_scheduling_update(self, mocker, support_update, instance_type, match):
+        plugin_old_configuration = f"""
+Image:
+  Os: alinux2
+  CustomAmi: ami-08cf50b131bcd4db2
+HeadNode:
+  InstanceType: t2.micro
+  Networking:
+    SubnetId: subnet-08a5068070f6bc23d
+  Ssh:
+    KeyName: ermann-dub-ef
+  Iam:
+    AdditionalIamPolicies:
+      - Policy: arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess
+Scheduling:
+  Scheduler: plugin
+  SchedulerSettings:
+    SchedulerDefinition:
+      PluginInterfaceVersion: "1.0"
+      Requirements:
+        SupportsClusterUpdate: {support_update}
+      Events:
+        HeadInit:
+          ExecuteCommand:
+            Command: env
+  SchedulerQueues:
+    - Name: queue1
+      Networking:
+        SubnetIds:
+          - subnet-12345678
+      ComputeResources:
+        - Name: compute-resource1
+          InstanceType: c5.2xlarge
+"""
+
+        plugin_new_configuration = f"""
+Image:
+  Os: alinux2
+  CustomAmi: ami-08cf50b131bcd4db2
+HeadNode:
+  InstanceType: t2.micro
+  Networking:
+    SubnetId: subnet-08a5068070f6bc23d
+  Ssh:
+    KeyName: ermann-dub-ef
+  Iam:
+    AdditionalIamPolicies:
+      - Policy: arn:aws:iam::aws:policy/FakePolicy
+Scheduling:
+  Scheduler: plugin
+  SchedulerSettings:
+    SchedulerDefinition:
+      PluginInterfaceVersion: "1.0"
+      Requirements:
+        SupportsClusterUpdate: {support_update}
+      Events:
+        HeadInit:
+          ExecuteCommand:
+            Command: env
+  SchedulerQueues:
+    - Name: queue1
+      Networking:
+        SubnetIds:
+          - subnet-12345678
+      ComputeResources:
+        - Name: compute-resource1
+          InstanceType: {instance_type}
+"""
+
+        mock_aws_api(mocker)
+        cluster = Cluster(
+            FAKE_NAME,
+            stack=ClusterStack(
+                {
+                    "StackName": FAKE_NAME,
+                    "CreationTime": "2021-06-04 10:23:20.199000+00:00",
+                    "StackStatus": ClusterStatus.CREATE_COMPLETE,
+                }
+            ),
+            config=plugin_old_configuration,
+        )
+
+        mocker.patch("pcluster.aws.cfn.CfnClient.stack_exists", return_value=True)
+
+        if match:
+            with pytest.raises(ClusterUpdateError, match=match):
+                cluster.validate_update_request(
+                    target_source_config=plugin_new_configuration, validator_suppressors={AllValidatorsSuppressor()}
+                )
```

Comment:
shall we also assert the changes?

Reply:
Added the assertion for scheduling_changes